R-package/tests/testthat/test-RGF_package.R

```diff
@@ -288,113 +283,257 @@ testthat::test_that("the 'mat_2scipy_sparse' returns a scipy CSC sparse matrix",
+
+  init_class = RGF_Classifier$new(max_leaf = 50, sl2 = 0.1, n_iter = 10)
+  init_class$fit(x = x_rgf, y = y_BINclass)
+  
+  lst_files_tmp_upd_rgf = list.files(path = default_dir)
+  
+  init_exists_upd_rgf = (dir.exists(default_dir) == TRUE)
+  init_num_files_rgf = length(lst_files_tmp_upd_rgf)
+  
+  init_class$cleanup()
+  
+  lst_files_tmp_upd_rgf = list.files(path = default_dir)
+  init_num_files_rgf_after_clean = length(lst_files_tmp_upd_rgf)
+  
+  end_state_rgf = (init_num_files_rgf_after_clean < init_num_files_rgf)
+  
+  #-------------------------------------------------------------------------------- FastRGF
+  init_class = FastRGF_Classifier$new(n_estimators = 50, max_bin = 65000)
+  init_class$fit(x = x_FASTrgf, y = y_MULTIclass)
+  
+  lst_files_tmp_upd_fastrgf = list.files(path = default_dir)
+  init_num_files_fastrgf = length(lst_files_tmp_upd_fastrgf)
+  
+  init_class$cleanup()
+  
+  lst_files_tmp_upd_fastrgf = list.files(path = default_dir)
+  init_num_files_fastrgf_after_clean = length(lst_files_tmp_upd_fastrgf)
+  
+  end_state_fastrgf = (init_num_files_fastrgf_after_clean < init_num_files_fastrgf)
+  
+  testthat::expect_true( init_exists_upd_rgf && end_state_rgf && end_state_fastrgf )
+})
+
+
+
+testthat::test_that("the 'cleanup' method (APPLIES TO ALL ESTIMATORS) works as expected for both RGF and FastRGF (checking of the size of the temporary directory before and after the '$fit' method)", {
+  
+  skip_test_if_no_module("rgf.sklearn")
+  
+  #-------------------------------------------------------------------------------- default directory where the temporary 'rgf' files are saved 
+  
+  default_dir = file.path(dirname(tempdir()), 'rgf')
+  
+  #-------------------------------------------------------------------------------- RGF
+  lst_files_tmp = list.files(path = default_dir)
+
+  init_class = RGF_Classifier$new(max_leaf = 50, sl2 = 0.1, n_iter = 10)
+  init_class$fit(x = x_rgf, y = y_BINclass)
+  
+  lst_files_tmp_upd_rgf = list.files(path = default_dir)
+  
+  init_exists_upd_rgf = (dir.exists(default_dir) == TRUE)
+  init_num_files_rgf = length(lst_files_tmp_upd_rgf)
+
+  #-------------------------------------------------------------------------------- FastRGF
+  init_class = FastRGF_Classifier$new(n_estimators = 50, max_bin = 65000)
+  init_class$fit(x = x_FASTrgf, y = y_MULTIclass)
+  
+  lst_files_tmp_upd_fastrgf = list.files(path = default_dir)
+  init_num_files_fastrgf = length(lst_files_tmp_upd_fastrgf)
+  
+  RGF_cleanup_temp_files()
+  
+  lst_files_tmp_end_state = list.files(path = default_dir)
+  
+  testthat::expect_true( init_exists_upd_rgf && (init_num_files_rgf > 0) && (init_num_files_fastrgf > init_num_files_rgf) &&
+                            ( init_num_files_rgf > length(lst_files_tmp) && init_num_files_fastrgf > length(lst_files_tmp_end_state) ) )       # normally, both initial and end state must have the same length [ length(lst_files_tmp) == length(lst_files_tmp_end_state) ]
```

Comment:
I guess the last check can be more strict: `init_num_files_rgf > length(lst_files_tmp_end_state)` instead of the current one `init_num_files_fastrgf > length(lst_files_tmp_end_state)`.

Reply:
@StrikerRUS, 
for this test I'd like yours or @fukatani's guidance too. 

Initially, when I started writing the test I thought of the following,

- I start from an empty temporary directory ('/tmp/rgf') and after running an estimator, say ```RGF_Classifier```, a specific number of temporary files will be created, say 5 in total. Then, I run ```fit.cleanup()``` and those specific to the estimator temporary files will be removed. Am I correct on this? What if the '/tmp/rgf' directory has files from a previous estimator, will be only the last estimator's temporary files be removed? You can see also my commented out line 

```normally, both initial and end state must have the same length [ length(lst_files_tmp) == length(lst_files_tmp_end_state) ]```
- from the [python test cases](https://github.com/RGF-team/rgf/blob/master/python-package/tests/test_rgf_python.py#L39-L82) I observed that there is actually no counting of already created temporary files. Do I have to do the same in the R tests? Let me know what you think.

Reply:
> Then, I run fit.cleanup() and those specific to the estimator temporary files will be removed. Am I correct on this?

I think you're absolutely correct. And yes, it's possible that files from previous estimators locate there. Ideally, all files from all **created** estimators are cleaned up during Python module unloading event, refer to
https://github.com/RGF-team/rgf/blob/44b3654b5801273bc1a2e0d291a734f43d6b0b48/python-package/rgf/utils.py#L34-L37

Reply:
@StrikerRUS, to clarify my intentions,

- when I use the ```estimator.cleanup()``` method I expect that all related to this estimator temporary files are removed, regardless if there are previous files (from another estimator) in the same directory. 
- when I use the ```cleanup()``` method (global removal of files, which means that all files in the rgf temporary directory will be removed) I expect that before fitting any estimators and after the cleanup the directory is empty.

Can you please confirm that this is the case in python and let me know, so that I can adjust the R tests? thanks.

Reply:
Yes, it is supposed to be so.

> I expect that before fitting any estimators and after the cleanup the directory is empty.

Sometimes directory is not empty before fitting any estimator due to temp files from previous session. They are there because of `@atexit.register` failure in case of any errors during previous user session. However, I don't think this is the case for test suit because temp folder is guaranteed to be clean.

Reply:
Nothing to do here as the ```init_num_files_rgf > length(lst_files_tmp_end_state)``` already exists in the test check.